umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif10.clj

```diff
@@ -113,9 +140,7 @@
         (util/remove-map-keys empty?
                               {:PrecisionOfSeconds (value-of temporal "Precision_Of_Seconds")
                                :EndsAtPresentFlag (value-of temporal "Ends_At_Present_Flag")
-                               :RangeDateTimes (for [rdt (select temporal "Range_DateTime")]
-                                                 {:BeginningDateTime (date/with-default (value-of rdt "Beginning_Date_Time") sanitize?)
-                                                  :EndingDateTime (parse-dif-end-date (value-of rdt "Ending_Date_Time"))})
+                               :RangeDateTimes (parse-range-date-times temporal sanitize?)
```

Comment:
Nice -- this is a great clean-up pattern we need to do more of: pulling nested logic out into one or more functions.